### PR TITLE
add py-vl-convert-python

### PR DESCRIPTION
**Pull request by wu-kan**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-vl-convert-python/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    version("0.4.0", sha256="12d56ff4b137e058b6f8e1ada9b06609b109c084c7060b2d1b0548ffac1898d6")
+    version("0.3.2", sha256="a0b0827827d727a4c8d9a2989d5cfd11ba52cb9f0ff450a0dd6df33326922662")
+    version("0.3.1", sha256="01c8ae5edd66bbcf097437a2bcb1c7be7aeb262a3c55648cfabcf666c26a3850")
+    version("0.3.0", sha256="2a76ac70d2ef9ad772602a9559f453d95fa9297facefabdbbfaca86372522f6d")
+    version("0.2.0", sha256="92df145b8f232962d22141234630054ba05d7066b2ca262d0f10c671728b9b0b")
+    version("0.1.0", sha256="92875d9ca02d1b7683cde9a92c5a910bb7406c013d38fdf8564ffd69907a4529")
+    version("0.0.1", sha256="cc6d922b734e25bc8a314c0c01d7744c35d87ed3095a6377299b65a2b79fd946")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-maturin", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-vl-convert-python/package.py]:
```suggestion
    depends_on("py-maturin@1.1:1", type="build")

```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Do we also need to add the dependencies from `Cargo.toml` to avoid pip downloading things during the install?

**Comment by wu-kan** [comment]
I tried, but failed. And I find other packages also suffer from this (e.g., eza, py-tokenizers). Maybe we need a new `download_dependencies` phases in `spack.build_systems.cargo` and `spack.build_systems.python_maturin` …

**Comment by wu-kan** [comment]
> Do we also need to add the dependencies from `Cargo.toml` to avoid pip downloading things during the install?

I tried, but failed. And I find other packages also suffer from this (e.g., eza, py-tokenizers). Maybe we need a new `download_dependencies` phases in `spack.build_systems.cargo` and `spack.build_systems.python_maturin` …

**Comment by wu-kan** [comment]
Just for the `py-vl-convert-python` package, another solution is using bdists (whl) over sdists (tar.gz).

File: var/spack/repos/builtin/packages/py-vl-convert-python/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    pypi = "vl_convert_python/vl_convert_python-0.13.1.tar.gz"
+
+    version("0.13.1", sha256="d70a608257dd6b5b782d96cccebfe7289992e522e47a8bebb7d928253ca8b396")
+
+    depends_on("python@3.7:", type=("build", "run"))
+
+    # TODO: This package currently requires internet access to install.
+    depends_on("py-maturin@1.1:1", type="build")
+
+    depends_on("cmake", type="build")  # some rust dependcies need this
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-vl-convert-python/package.py]:
```suggestion
    depends_on("cmake", type="build")  # some rust dependencies need this

```

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**